### ISSUE TITLE
Timeline: Custom styles on items break selection

### ISSUE THREAD

**Issue by gossi** [almende/vis#2455]
When items on a Timeline vis have a custom style property given, the selection won't work. Basically, when you select an item, the positional properties (e.g. `left`) is overwridden with the custom style rsp. the positional style isn't applied. As a result, when you select an item on a timeline with custom style the item is positioned left. If you move it around or change start or end, it still works as expected with the original positon. As though, it only affects the selecting process.

**Comment by yotamberk** [comment]
@gossi Can you please supply a jsbin so we can see the problem you're encountering?

**Comment by yotamberk** [comment]
@gossi I don't quite understand why you would try to change the `left` property. Its position should be derived by the time it has and should not be changed externally by css.

**Comment by gossi** [comment]
No, I don't. My custom `style` is color and `background-color`. Yet they overwrite the `left` property that the timeline gives these items.

Is there a template for jsbin?

**Comment by yotamberk** [comment]
@gossi no template but there are many jsbin examples in other issues. Try using them.

**Comment by gossi** [comment]
Ok, I tried to reproduce the problem on jsbin here: https://jsbin.com/ditonus/edit?js,output
Well, it works on jsbin o_O. What's relevant there is, you can see my used options and the item code, etc. So, I decided to take some screenshots from my problem. I took a screenshot from the timeline and the selected DOM element from the devtools. I took a series of 3 screenshot pairs:
(1) just the item on the timeline (2) the selected item on the timeline (3) click the timeline (to deselect the item)
Once you move the timeline, the item will be positioned correctly again. What's interessting here is the missing  `left` property on the DOM element, after selecting the item.

1) Timeline "as is"
<img width="776" alt="timeline-pre" src="https://cloud.githubusercontent.com/assets/283700/21463268/f3302b0c-c966-11e6-8c17-bcfb9447836a.png">
<img width="273" alt="code-pre" src="https://cloud.githubusercontent.com/assets/283700/21463270/f8f31cf2-c966-11e6-9d36-df71ab3f5e93.png">

2) Selecting the item
<img width="776" alt="timeline-selected" src="https://cloud.githubusercontent.com/assets/283700/21463272/05416a54-c967-11e6-971c-1ae1cd62382a.png">
<img width="272" alt="code-selected" src="https://cloud.githubusercontent.com/assets/283700/21463274/0b566228-c967-11e6-8db5-dc215b370d88.png">

3) Clicking the timeline/deselecting the item
<img width="775" alt="timeline-post" src="https://cloud.githubusercontent.com/assets/283700/21463278/1802a07c-c967-11e6-8ef8-40fdf81f4970.png">
<img width="271" alt="code-post" src="https://cloud.githubusercontent.com/assets/283700/21463279/19da9db4-c967-11e6-8f0f-8b93501b9a72.png">

Was the best I could provide at the moment. Let me know, if I can provide something else.

**Comment by gossi** [comment]
I recently updated to 4.18, secretly hoping the problem resolves itself. Unfortunately it got worse :/ With 4.18 the selection problem I outlined above still persists but additionally, my custom styles aren't applied either. So my fancy-colored-items persist in the same color as before.

**Comment by yotamberk** [comment]
@gossi 
Can you please supply an updated jsbin with the problem?
I'm not managing to reproduce this...

**Comment by gossi** [comment]
It is still the same code. However, I run it inside electron (ember-electron to be clear). So I guess it must also has something to do with that environment. Shall I try to create a minimum viable version of that?

**Comment by yotamberk** [comment]
yes please. I cannot recreate this scenario... A jsbin will be useful

**Comment by mojoaxel** [comment]
I'm closing this issue due to inactivity. 
If this issue is still relevant please feel free to reopen.

**Comment by gossi** [comment]
The problem isn't gone, though I have further insights.

First, here is a MWE demonstrating the problem: https://github.com/gossi/ember-electron-vis-timeline-problem

Checkout and run `yarn start` or `npm start` I dunno if it picks up the local binaries, if not it is `./node_modules/.bin/ember electron` to start this. This will pop up an electron window.

I made a simple timeline, with four items, two of them work the other two are broken. I outlined the problem for a 'range' item above, the same applies to the 'box' item as well, but without the additional style attribute.

I traced it down to the `redraw` method of `BoxItem`. To be exactly, when the temporary saved size is set back onto the dom box element, it won't be set (or electron/chrome doesn't pick up that one). It is this line:
https://github.com/almende/vis/blob/5563fc616885578ad51b2dfa6ccf70ccebf06126/lib/timeline/component/item/BoxItem.js#L171

I set a breakpoint here and stepped through: https://github.com/almende/vis/blob/5563fc616885578ad51b2dfa6ccf70ccebf06126/lib/timeline/component/item/BoxItem.js#L198

1) At first sizes are retrieved correctly
2) box.left is set to `0px` and moved to the left
3) box.left is set back to the `sizes.left` which is the problem. On the elements tab, you can see, the `left` property in the `style=""` attribute is correct, while in the CSS Rule tree on `Element` the `left` property is evaluated to `0`.

I would expect something similar happening for the `RangeItem` (though I didn't stepped through it).
